tools/timeit.c

```diff
@@ -140,10 +170,92 @@ static void timeout_handler(int signal) {
   kill(-g_monitored_pid, SIGKILL);
 }
 
+#define set_resource_limit(resource, value)                                    \
+  set_resource_limit_actual(#resource, resource, value)
+static void set_resource_limit_actual(const char *resource_name, int resource,
+                                      rlim_t value) {
+  /* Get the current limit. */
+  struct rlimit current;
+  getrlimit(resource, &current);
+
+  /* Set the limits to as close as requested, assuming we are not super-user. */
+  struct rlimit requested;
+  requested.rlim_cur = requested.rlim_max =
+      (value < current.rlim_max) ? value : current.rlim_max;
+  if (setrlimit(resource, &requested) < 0) {
+    fprintf(stderr, "%s: warning: unable to set limit for %s (to {%lu, %lu})\n",
+            g_program_name, resource_name, (unsigned long)requested.rlim_cur,
+            (unsigned long)requested.rlim_max);
+  }
+}
+#else
+
+/* Windows-specific implementation of fopen. */
+static FILE *windows_fopen(const char *filename, const char *mode) {
+  FILE *file;
+  if (fopen_s(&file, filename, mode))
+    return NULL;
+  return file;
+}
+#define fopen(filename, mode) windows_fopen(filename, mode)
+
+/* Process information struct global variable. */
+PROCESS_INFORMATION proc_info = {0};
```

Comment:
As other globals in the file seems to be using the naming convention of a `g_` prefix for globals, we could probably use that here as well. And it might be good to point out that the code generally uses local variables with `pid_t` for passing the reference to a process around, but here we're cheating with global variables instead, which only allows one single child process.

Reply:
proc_info is no longer a global after latest changes.

Reply:
Also commented about single threaded behaviour of the app in recent changes.